pkg/prometheus/agent/common.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2023 The prometheus-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheusagent
+
+import (
+	v1 "k8s.io/api/core/v1"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/prometheus-operator/prometheus-operator/pkg/operator"
+	prompkg "github.com/prometheus-operator/prometheus-operator/pkg/prometheus"
+)
+
+func buildImagePath(
+	cpf monitoringv1.CommonPrometheusFields,
+	c *prompkg.Config,
+) (string, error) {
+	pImagePath, err := operator.BuildImagePath(
+		operator.StringPtrValOrDefault(cpf.Image, ""),
+		operator.StringValOrDefault("", c.PrometheusDefaultBaseImage),
+		operator.StringValOrDefault(cpf.Version, operator.DefaultPrometheusVersion),
+		"",
+		"",
+	)
+
+	if err != nil {
+		return "", err
+	}
+
+	return pImagePath, nil
+}
+
+func buildAgentArgs(
+	cpf monitoringv1.CommonPrometheusFields,
+	cg *prompkg.ConfigGenerator,
+) []monitoringv1.Argument {
+	promArgs := prompkg.BuildCommonPrometheusArgs(cpf, cg)
+	promArgs = appendAgentArgs(promArgs, cg, cpf.WALCompression)
+
+	return promArgs
+}
+
+func buildVolumes(
+	p monitoringv1.PrometheusInterface,
+	tlsSecrets *operator.ShardedSecret,
+) ([]v1.Volume, []v1.VolumeMount, []v1.VolumeMount, error) {
+	volumes, promVolumeMounts, err := prompkg.BuildCommonVolumes(p, tlsSecrets)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	configReloaderVolumeMounts := []v1.VolumeMount{
```

Comment:
I see that we have the same snippet between Agent and Server. Should we rather create a `prompkg.ConfigReloaderVolumeMounts()`? Or even move it to `operator.CreateConfigReloader()`?

Reply:
same above

Reply:
yeah this makes sense I'll do it

Reply:
I'll move this to pkg/operator/config_reloader.go

Reply:
Now reread the code I'll move it to pkg/prometheus/statefulset.go instead, becaue we use prompkg fields here.

Reply:
Now reread the code I see we already had `pkg/prometheus/statefulset.go`, so I'll move it here.